File: PersisteniadeObjetos/Ej1/main.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

class Empleado implements Serializable {
    private String nombre;
    private int edad;
    private float salario;

    public Empleado(String nombre, int edad, float salario) {
        this.nombre = nombre;
        this.edad = edad;
        this.salario = salario;
    }

    public String getNombre() {
        return nombre;
    }

    public float getSalario() {
        return salario;
    }

    @Override
    public String toString() {
        return "Empleado: " + nombre + ", Edad: " + edad + ", Salario: " + salario;
    }
}

class ArchivoEmpleado {
    private String nomA;

    public ArchivoEmpleado(String nomA) {
        this.nomA = nomA;
    }

    public void crearArchivo() {
        try {
            new File(nomA).createNewFile();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public void guardarEmpleado(Empleado e) {
        List<Empleado> empleados = leerEmpleados();
        empleados.add(e);
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(nomA))) {
            oos.writeObject(empleados);
        } catch (IOException ex) {
            ex.printStackTrace();
        }
    }

    private List<Empleado> leerEmpleados() {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(nomA))) {
            return (List<Empleado>) ois.readObject();
        } catch (EOFException e) {
            return new ArrayList<>();
        } catch (IOException | ClassNotFoundException ex) {
            ex.printStackTrace();
            return new ArrayList<>();
        }
    }

    public Empleado buscaEmpleado(String nombre) {
        List<Empleado> empleados = leerEmpleados();
        for (Empleado e : empleados) {
            if (e.getNombre().equalsIgnoreCase(nombre)) {
                return e;
            }
        }
        return null;
    }

    public Empleado mayorSalario(float sueldo) {
        List<Empleado> empleados = leerEmpleados();
        for (Empleado e : empleados) {
            if (e.getSalario() > sueldo) {
                return e;
            }
        }
        return null;
    }

    public static void main(String[] args) {
        ArchivoEmpleado archivo = new ArchivoEmpleado("empleados.dat");
        archivo.crearArchivo();

        archivo.guardarEmpleado(new Empleado("Ana", 30, 2500));
        archivo.guardarEmpleado(new Empleado("Luis", 40, 3200));
        archivo.guardarEmpleado(new Empleado("Pedro", 35, 2800));
        archivo.guardarEmpleado(new Empleado("Maria", 28, 3000));
        archivo.guardarEmpleado(new Empleado("Juan", 45, 3500));

        System.out.println("Buscado: " + archivo.buscaEmpleado("Luis"));
        System.out.println("Mayor salario: " + archivo.mayorSalario(2500));
    }
}

File: PersisteniadeObjetos/Ej1/main.py
import csv

class Empleado:
    def __init__(self, nombre: str, edad: int, salario: float):
        self.nombre = nombre
        self.edad = edad
        self.salario = salario

    def __str__(self):
        return f"Empleado: {self.nombre}, Edad: {self.edad}, Salario: {self.salario}"

class ArchivoEmpleado:
    def __init__(self, nom_archivo: str):
        self.nom_archivo = nom_archivo

    def crear_archivo(self):
        try:
            with open(self.nom_archivo, 'x') as file:
                pass  # Crea el archivo vacío
        except FileExistsError:
            pass  # El archivo ya existe

    def guardar_empleado(self, empleado: Empleado):
        with open(self.nom_archivo, 'a', newline='') as file:
            writer = csv.writer(file)
            writer.writerow([empleado.nombre, empleado.edad, empleado.salario])

    def buscar_empleado(self, nombre: str) -> Empleado:
        with open(self.nom_archivo, 'r') as file:
            reader = csv.reader(file)
            for row in reader:
                if row[0].lower() == nombre.lower():
                    return Empleado(row[0], int(row[1]), float(row[2]))
        return None

    def mayor_salario(self, sueldo_min: float) -> Empleado:
        with open(self.nom_archivo, 'r') as file:
            reader = csv.reader(file)
            for row in reader:
                if float(row[2]) > sueldo_min:
                    return Empleado(row[0], int(row[1]), float(row[2]))
        return None

archivo = ArchivoEmpleado("empleados.txt")
archivo.crear_archivo()

# Guardar empleados
archivo.guardar_empleado(Empleado("Ana", 30, 2500.0))
archivo.guardar_empleado(Empleado("Luis", 40, 3200.0))
archivo.guardar_empleado(Empleado("Carlos", 35, 2800.0))
archivo.guardar_empleado(Empleado("María", 28, 3000.0))


# Buscar empleado
buscado = archivo.buscar_empleado("Luis")
print(f"Empleado encontrado: {buscado}")

# Mayor salario
mejor_pagado = archivo.mayor_salario(2800.0)
print(f"Empleado con mayor salario: {mejor_pagado}")